### PR TITLE
LJ-89 Side Menu

### PR DESCRIPTION
**Pull request by MarcoGoC**
# Description

[LJ-89] Side sticky menu

created side menu 
changes to the having a child page as an examaple to be followed added for other pages.

## Test Instructions

1. Get the branch and see it sticks to the top when navigating 
1. Anchors to Open the accordion are in place but it requires extra js coding to open using the menu, this part is not implemented on this branch but as a separate task.

## Meets Definition of Done

- [ ] Meets design spec
- [ ] unit tests are included

[LJ-89]: https://lj-decd.atlassian.net/browse/LJ-89

### CODE REVIEW

File: components/atoms/Accordion.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function Accordion(props) {
   const arrowColor = "#D3080C";
 
   return (
-    <div id="accordion" className="flex mb-2 container">
+    <div id="accordion" className="flex mb-2">
```
Comment (krischarbonneau) [inline components/atoms/Accordion.js:16]: id needs to be unique. The way it is now there are 7 Accordions with the same id.

File: components/atoms/SideMenu.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export function SideMenu({ title, links }) {
+  return (
+    <div
+      id="Sidemenu"
+      className="mt-2 p-4 sticky top-0 block max-w-sm float-left text-white"
+    >
+      <h2 className="font-bold text-p mb-2">{title}</h2>
+
+      <nav className="mt-2 mb-2 mr-4 block">
```
Comment (krischarbonneau) [inline components/atoms/SideMenu.js:12]: Add ```id="side-nav" aria-label="SideMenu"``` here so this nav landmark is unique

**Review by krischarbonneau — COMMENTED** [review]
I think this looks really good as is, but on mobile resolutions this should either hide or not stick to the side of the screen. Couple other accessibility changes suggested.

**Comment by MarcoGoC** [comment]
I don't think we have decided what to do on mobile

**Comment by krischarbonneau** [comment]
> I don't think we have decided what to do on mobile

Thought we decided that it stick to the top of the screen on mobile for now with the intention of hiding it in the future (since that is not MVP)

File: styles/globals.css [hunk 3]
```diff
@@ -639,6 +639,38 @@ ul, ol {
...
+  display: inline-flex;
+  flex-direction: column;
+  justify-content: space-between; 
+  padding: 2px 12px 2px 0px;
+  color: black;
+  text-decoration: none;
+}
+.menuLink::after {
+  content: attr(data-text);
+  content: attr(data-text)/" ";
```
Comment (pcraig3) [inline styles/globals.css:655]: Only one of these will get used -- the other will be overridden.

File: pages/having-a-child.js [hunk 4]
```diff
@@ -23,14 +24,101 @@ export default function lifejourney() {
...
+        <div className="w-3/12 sticky top-0">
+          <SideMenu />
+        </div>
+
+        <div className="w-9/12 float-right">
+          <Accordion
+            id="GettingReady"
+            title="Getting Ready"
+            summary="Getting Ready summary text"
+          >
```
Comment (pcraig3) [inline pages/having-a-child.js:38]: I've noticed the accordions all have an `id="accordion"`, which we should remove. IDs are supposed to be unique on the page, so better to not have one.

**Review by pcraig3 — CHANGES_REQUESTED** [review]
I think the code looks really good here, but there are a few visual tweaks we should do to make this look a bit better.

| on Desktop | on Mobile |
|--------|-------|
|   Screen Shot 2021-05-04 at 09 51 37    |  Screen Shot 2021-05-04 at 09 51 48   |
|   Once you squish the screen a little, the blue menu touches the accordions. On this screen, we should add some right margin to the accordion column, and some bottom margin to the breadcrumbs as well.      |  On mobile, we should remove `sticky` and just have the menu be full width above all the accordions.   |

File: pages/having-a-child.js (hunk 4 above)
Comment (MarcoGoC) [reply pages/having-a-child.js:38]: done already, the id on the accordion becomes the anchor for the side menu links

File: styles/globals.css (hunk 3 above)
Comment (MarcoGoC) [reply styles/globals.css:655]: i found this solution somewhere i am guessing there is a reason for it, maybe old/specific browser.

**Review by pcraig3 — APPROVED** [review]
Looks good to me — probably will want to change this design one day but without designers on the team, I think this is a pretty good baseline.

Thanks @MarcoGoC!

File: pages/having-a-child.js [hunk 5]
```diff
@@ -23,14 +24,101 @@ export default function lifejourney() {
...
         <Breadcrumb />
       </div>
-      <Accordion>
-        <div className="pr-6">
-          <MoreInfo text={t.moreInfoCommunity} />
-          <MoreInfo text={t.moreInfoFertility} />
-          <MoreInfo text={t.moreInfoHealth} />
-          <MoreInfo text={t.moreInfoFinancial} />
+
+      <div className="container flex flex-col md:flex-row align-items-center ">
```
Comment (krischarbonneau) [inline pages/having-a-child.js:28]:
```suggestion
      <div className="container flex flex-col md:flex-row align-items-center items-start ">
```
The div on line 29 is taking up the entire height of the parent div on line 28, if we add ```items-start``` to the end here makes it so each of the children are only the height of their content. This makes the sticky function work as intended as larger resolutions.

File: pages/having-a-child.js [hunk 6]
```diff
@@ -23,14 +24,101 @@ export default function lifejourney() {
...
+        <div className="pr-3 w-full lg:w-3/12 top-0 lg:sticky ">
```
Comment (krischarbonneau) [inline pages/having-a-child.js:29]:
```suggestion
        <div className="pr-3 w-full lg:w-3/12 top-0 md:sticky ">
```
The side menu still floats to the left of the accordions at medium resolutions, maybe make it sticky at those resolutions as well? The menu container itself has a lot of extra space beside the 'quick links' at this res but I don't think that's something we need to worry a ton about at this stage.

**Review by krischarbonneau — APPROVED** [review]
Looks great, good job Marco :)

**Pull request merged**